### PR TITLE
Respect distribution-provided properties

### PR DESCRIPTION
**Pull request by labkey-adam**
#### Rationale
As of the newest gradle plugins version, distributions now include a properties file with distribution name, archive filename, and (optionally) supported databases. This PR reads and uses those properties.

Related: https://www.labkey.org/home/Developer/issues/Secure/issues-details.view?issueId=51326

#### Related Pull Requests
- https://github.com/LabKey/gradlePlugin/pull/219

#### Changes
- Read `distribution.properties` file, if present. Read it once & stash the results and optimize `shouldReceiveMarketingUpdates()` to address [Issue 51326](https://www.labkey.org/home/Developer/issues/Secure/issues-details.view?issueId=51326).
- Stop reading standalone `distribution` file
- Show distribution name and filename on server information page and export diagnostics
- In production mode, throw if the primary database is not supported by the distribution
- Stop allowing `cpasDataSource`
- Deprecate support for PostgreSQL 12 and start claiming that PostgreSQL 17 (just released) is tested

### CODE REVIEW

File: api/src/org/labkey/api/module/SupportedDatabase.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    public static SupportedDatabase get(SqlDialect dialect)
+    {
+        if (dialect.isPostgreSQL())
+            return pgsql;
+
+        if (dialect.isSqlServer())
+            return mssql;
+
+        throw new IllegalStateException("Dialect not supported");
```
Comment (labkey-jeckels) [inline api/src/org/labkey/api/module/SupportedDatabase.java]: Unlikely to ever hit this, but always nice to include the dialect that was supplied in the error message

File: api/src/org/labkey/api/util/MothershipReport.java [hunk 2]
```diff
@@ -463,7 +456,8 @@ public void addServerSessionParams()
         ServletContext context = ModuleLoader.getServletContext();
         String servletContainer = context == null ? null : context.getServerInfo();
         addParam("servletContainer", servletContainer);
-        addParam("distribution", getDistributionName());
+        addParam("distribution", AppProps.getInstance().getDistributionName());
+        addParam("distributionFilename", AppProps.getInstance().getDistributionFilename());
```
Comment (labkey-jeckels) [inline api/src/org/labkey/api/util/MothershipReport.java]: These values are stored in real columns in `mothership.serversession`. Better to add this to the JSON metrics, likely in `CoreModule`. As-is, the `distributionFilename` value will be ignored.

**Review by labkey-jeckels — APPROVED** [review]

**Review by labkey-tchad — CHANGES_REQUESTED** [review]
`EmbeddedExtractor` needs to be updated to handle the new `distribution.properties` file. `shouldExtract`, `getDistributionInfo`, and `LabKeyDistributionInfo` need to know the name of the new file and how to handle the new structure

**Comment by labkey-adam** [comment]
> `EmbeddedExtractor` needs to be updated to handle the new `distribution.properties` file. `shouldExtract`, `getDistributionInfo`, and `LabKeyDistributionInfo` need to know the name of the new file and how to handle the new structure

Yep, I have those changes locally

**Comment by labkey-adam** [comment]
> > `EmbeddedExtractor` needs to be updated to handle the new `distribution.properties` file. `shouldExtract`, `getDistributionInfo`, and `LabKeyDistributionInfo` need to know the name of the new file and how to handle the new structure
> 
> Yep, I have those changes locally

@labkey-tchad those changes need to go into a separate PR since they're in the `server` repo. Can you approve this PR to unblock it?

**Review by labkey-tchad — APPROVED** [review]

**Pull request merged**